jax_galsim/gsparams.py

```diff
@@ -0,0 +1,123 @@
+import copy
+from dataclasses import dataclass
+import galsim as _galsim
+from jax._src.numpy.util import _wraps
+
+
+@_wraps(_galsim.GSParams)
+@dataclass(frozen=True, repr=False)
+class GSParams:
+
+    minimum_fft_size: int = 128
+    maximum_fft_size: int = 8192
+    folding_threshold: float = 5.0e-3
+    stepk_minimum_hlr: float = 5
+    maxk_threshold: float = 1.0e-3
+    kvalue_accuracy: float = 1.0e-5
+    xvalue_accuracy: float = 1.0e-5
+    table_spacing: int = 1
+    realspace_relerr: float = 1.0e-4
+    realspace_abserr: float = 1.0e-6
+    integration_relerr: float = 1.0e-6
+    integration_abserr: float = 1.0e-8
+    shoot_accuracy: float = 1.0e-5
+
+    @staticmethod
+    def check(gsparams, default=None, **kwargs):
+        """Checks that gsparams is either a valid GSParams instance or None.
+
+        In the former case, it returns gsparams, in the latter it returns default
+        (GSParams.default if no other default specified).
+        """
+        if gsparams is None:
+            gsparams = default if default is not None else GSParams.default
```

Comment:
hum it will return a TypeError("Invalid GSParams: %s" % gsparams) if the default is not a GSParams. I think this should be ok as this function is directly stolen from galsim

Reply:
> hum it will return a TypeError("Invalid GSParams: %s" % gsparams)
I think you won't get to return this if `gsparams == None` and `default` is not None but something that is not gsparams. Because you won't get to run the `elif` block. 

> this should be ok as this function is directly stolen from galsim
ok! then in that case it is probably good

Reply:
right, like:
```python
galsim.GSParams.check(None, default='toto')
```
indeed ^^ this fails. We can make a PR to GalSim.

Reply:
https://github.com/GalSim-developers/GalSim/pull/1152

Reply:
ah that's very nice actually, I think you can resolve this comment then